src/__swaps__/screens/Swap/components/SwapNumberPad.tsx

```diff
@@ -49,23 +51,78 @@ const getFormattedInputKey = (inputKey: inputKeys) => {
 
 export const SwapNumberPad = () => {
   const { isDarkMode } = useColorMode();
-  const { focusedInput, isQuoteStale, SwapInputController, configProgress, outputQuotesAreDisabled } = useSwapContext();
+  const { nativeCurrency } = useAccountSettings();
+  const {
+    focusedInput,
+    internalSelectedInputAsset,
+    internalSelectedOutputAsset,
+    isQuoteStale,
+    SwapInputController,
+    configProgress,
+    outputQuotesAreDisabled,
+  } = useSwapContext();
 
   const longPressTimer = useSharedValue(0);
 
-  const addNumber = (number?: number) => {
+  const removeFormatting = (inputKey: inputKeys) => {
     'worklet';
+    return stripNonDecimalNumbers(SwapInputController[getFormattedInputKey(inputKey)].value);
+  };
+
+  const ignoreChange = ({ currentValue, addingDecimal = false }: { currentValue?: string; addingDecimal?: boolean }) => {
+    'worklet';
+    // ignore when: outputQuotesAreDisabled and we are updating the output amount or output native value
     if ((focusedInput.value === 'outputAmount' || focusedInput.value === 'outputNativeValue') && outputQuotesAreDisabled.value) {
+      return true;
+    }
+
+    // ignore when: corresponding asset does not have a price and we are updating native inputs
+    const inputAssetPrice = internalSelectedInputAsset.value?.nativePrice || internalSelectedInputAsset.value?.price?.value || 0;
```

Comment:
this section makes sure that we only let the user update the native value if the corresponding asset has a price